pupil-spa/src/app/services/storage/storage.service.ts

```diff
@@ -54,4 +54,23 @@ export class StorageService {
       return obj;
     }, {});
   }
+
+  mergeItems(key: StorageKey): any {
+    const localStorageItems = this.getAllItems();
+    const matchingKeys =
+      Object.keys(localStorageItems).filter(lsi => lsi.indexOf(key) >= 0);
+    const sortedMatchingKeys = matchingKeys.sort((a, b) => localStorageItems[a].clientTimestamp - localStorageItems[b].clientTimestamp);
```

Comment:
This seems odd - I thought these values were strings... perhaps I am missing something?

Reply:
this is now using getTime() to compare the date values as we are always setting timestamps as `Date` structures